src/reputation/serializers.py

```diff
@@ -25,7 +44,54 @@
 )
 from user.models import Author
 
+class ProofRelatedField(serializers.RelatedField):
+    """
+    A custom field to use for the `source` generic relationship.
+    """
+
+    def to_representation(self, value):
+        """
+        Serialize tagged objects to a simple textual representation.
+        """
+        from paper.serializers import DynamicPaperSerializer, DynamicPaperVoteSerializer
+        from purchase.serializers import PurchaseSerializer
+
+        if isinstance(value, Comment):
+            return CommentSerializer(value).data
+        elif isinstance(value, Thread):
+            return ThreadSerializer(value).data
+        elif isinstance(value, Reply):
+            return ReplySerializer(value).data
+        elif isinstance(value, DisVote):
+            return DisVoteSerializer(value).data
+        elif isinstance(value, Paper):
+            paper_include_fields = [
+                'id',
+                'paper_title',
+                'slug',
+                'score'
+            ]
+            return DynamicPaperSerializer(value, _include_fields=paper_include_fields).data
+        elif isinstance(value, PaperVote):
+            return DynamicPaperVoteSerializer(
+                value,
+                _include_fields=['paper'],
+                context={
+                    'pap_dpvs_paper': {
+                        '_include_fields': ['id', 'paper_title', 'slug', 'score']
+                    }
+                }
+            ).data
+        elif isinstance(value, Purchase):
+            return PurchaseSerializer(value, context={'exclude_stats': True}).data
+
+        sentry.log_info('No representation for {} / id: {}'.format(str(value), value.id))
+        return None
+
+
 class DistributionSerializer(serializers.ModelSerializer):
+    proof_item = ProofRelatedField(read_only=True)
```

Comment:
I see this not being used. Is that intentional?

Reply:
this is being used -- it's returned with the serializer